### DISCUSSION TITLE
How to Disable Tooltips in Pyright for Vim

### DISCUSSION THREAD

**Discussion by GrassiColas** [microsoft/pyright#7790] [Q&A]
Hello, I’m trying to set up Pyright correctly, but I'm having trouble preventing this tooltip from appearing, as shown in the picture.

<img width="1326" alt="Capture d’écran 2024-04-27 à 23 14 17" src="https://github.com/microsoft/pyright/assets/122752724/55a80e4b-853c-4b38-a090-6e91e9d9c75e">

I've disabled all diagnostics, but this tooltip still keeps appearing.
Here's what I've tried in my coc-settings.json

```
{
  "coc.source.buffer.enable": true,
  "suggest.autoTrigger": "always",
  "semanticTokens.enable": true,
  "coc.preferences.semanticTokens": true,
  "coc.preferences.semanticTokensHighlights": true,
  "semanticTokens.filetypes": ["python"],
  "python.analysis.typeCheckingMode": "off",
  "python.analysis.diagnosticSeverityOverrides": {
    "*": "none"
}
```
so you cans see I use coc and coc-pyright(I only use this one)

**Comment by InSyncWithFoo** [comment]
Pyright on its own doesn't support inlay hint requests. These hints come from `coc-pyright` and can be customized using [its own configurations](https://github.com/fannheyward/coc-pyright#configurations). Add the following to your config file:

```json
{
  "pyright.inlayHints.functionReturnTypes": false,
  "pyright.inlayHints.variableTypes": false,
  "pyright.inlayHints.parameterTypes": false
}
```
